src/functions/function-command/build-and-deploy-function.ts

```diff
@@ -169,28 +158,59 @@ export async function buildFunction(context?: FunctionNode): Promise<CliExitData
   }
 }
 
-export async function deployFunction(context?: FunctionNode): Promise<void> {
-  const selectedFolderPick: FolderPick = await selectedFolder(context);
-  if (!selectedFolderPick && !context) {
+async function checkFuncIsBuild(context: FunctionNode): Promise<CliExitData> {
+  const funcYaml: string = await fs.readFile(path.join(context?.contextPath.fsPath, 'func.yaml'), 'utf-8');
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const getFuncYaml = yaml.safeLoadAll(funcYaml);
+  if (!getFuncYaml?.[0]?.image) {
+    const response: string = await vscode.window.showInformationMessage(
+      'Image not found in func.yaml. Do you want to build before deploy?',
+      'Build',
+      'Cancel',
+    );
+    if (response === 'Build') {
+      const result = await buildFunction(context);
+      if (result.error) {
+        return null;
+      }
+      return result;
+    }
     return null;
```

Comment:
if the func has already been built it returns a null, but, doing so, the deploy action is skipped https://github.com/redhat-developer/vscode-knative/pull/271/files#diff-f24a17d49f067029e4e7a483c4b9eed474aa9f14e68d8acf95ee11c758ba9257R186